src/v/pandaproxy/schema_registry/seq_writer.cc

```diff
@@ -134,9 +134,10 @@ void seq_writer::advance_offset_inner(model::offset offset) {
 
 ss::future<schema_id>
 seq_writer::write_subject_version(canonical_schema schema) {
-    auto do_write = [schema{std::move(schema)}](
+    auto do_write = [_schema{std::move(schema)}](
                       model::offset write_at,
                       seq_writer& seq) -> ss::future<std::optional<schema_id>> {
+        auto schema{_schema};
```

Comment:
I feel like we have a large number of violations of this rule in the code base?

Reply:
Depends how often parameters are used after the first `co_await`, but almost certainly there are many more. I think I got all the ones in schema registry.

Reply:
It's probably worth pointing out that this trick works because seastar awaitable are `std::suspend_never initial_suspend() noexcept`.

And for posterity: [But what if I told you there was an easier way, where you can have your capturing lambda be a coroutine?](https://devblogs.microsoft.com/oldnewthing/20211103-00/?p=105870#:~:text=but%20what%20if%20i%20told%20you%20there%20was%20an%20easier%20way%2C%20where%20you%20can%20have%20your%20capturing%20lambda%20be%20a%20coroutine%3F)

Reply:
These are the ones I could find: https://github.com/vectorizedio/redpanda/pull/3600